CMakeLists.txt

```diff
@@ -179,19 +242,27 @@ find_package(Sanitizers)
 
 # Strict c++17, e.g. no GNU extensions
 if (MSVC) # For MSVC, errors show up when using c++ 17 for build.
-  set_property(TARGET OpenSCAD PROPERTY CXX_STANDARD 20)
+  set_property(TARGET OpenSCADLibInternal PROPERTY CXX_STANDARD 20)
+  set_property(TARGET svg PROPERTY CXX_STANDARD 20)
 else()
-  set_property(TARGET OpenSCAD PROPERTY CXX_STANDARD 17)
+  set_property(TARGET OpenSCADLibInternal PROPERTY CXX_STANDARD 17)
+  set_property(TARGET svg PROPERTY CXX_STANDARD 17)
 endif()
-set_property(TARGET OpenSCAD PROPERTY CXX_EXTENSIONS OFF)
-set_property(TARGET OpenSCAD PROPERTY CXX_STANDARD_REQUIRED ON)
+set_property(TARGET OpenSCADLibInternal PROPERTY CXX_EXTENSIONS OFF)
+set_property(TARGET OpenSCADLibInternal PROPERTY CXX_STANDARD_REQUIRED ON)
 # Output compilation database (compile_commands.json), so we can e.g. run clang-tidy or other tools separately
-set_property(TARGET OpenSCAD PROPERTY EXPORT_COMPILE_COMMANDS ON)
-set_property(TARGET OpenSCAD PROPERTY LINKER_LANGUAGE CXX)
+set_property(TARGET OpenSCADLibInternal PROPERTY EXPORT_COMPILE_COMMANDS ON)
+set_property(TARGET OpenSCADLibInternal PROPERTY LINKER_LANGUAGE CXX)
+set_property(TARGET OpenSCADExe PROPERTY EXPORT_COMPILE_COMMANDS ON)
+set_property(TARGET OpenSCADExe PROPERTY LINKER_LANGUAGE CXX)
+set_property(TARGET svg PROPERTY EXPORT_COMPILE_COMMANDS ON)
+set_property(TARGET svg PROPERTY LINKER_LANGUAGE CXX)
```

Comment:
Here, and a few other places: This gets a bit repetitive. I believe you can do this:
```
set_target_properties(OpenSCADLibInternal svg OpenSCADExe PROPERTIES
                      EXPORT_COMPILE_COMMANDS ON
                      LINKER_LANGUAGE CXX)
```